shared/crypto/keypair.go

```diff
@@ -111,6 +115,16 @@ func (kp encKeyPair) GetSeed(passphrase string) ([]byte, error) {
 	return privKey.Seed(), nil
 }
 
+// Derives a new child key using the key as the master
+func (kp encKeyPair) DeriveChild(passphrase string, index uint32) (KeyPair, error) {
+	seed, err := kp.GetSeed(passphrase)
+	if err != nil {
+		return nil, err
+	}
+	path := fmt.Sprintf(PoktAccountPathFormat, index)
+	return DeriveChild(path, seed)
```

Comment:
Seeing this usage (and being confused by where it's defined) made me think that maybe we should move `slip.go` and `slip_test.go` into `shared/crypto/slip`? 

It might be premature, but I'm hoping to have all the packages we're building here to be importable (more modularly) by others in the future. Will definitely help with growth, adoption, etc